framework/DataObjects/DataSet.py

```diff
@@ -0,0 +1,1853 @@
+    rlz = self._convertFinalizedDataRealizationToDict(rlz, unpackXArray)
+    return rlz
+
+  def _getRealizationFromDataByValue(self,match, tol=1e-15, unpackXArray=False):
+    """
+      Obtains a realization from the data storage using the provided index.
+      @ In, match, dict, elements to match
+      @ In, tol, float, optional, tolerance to which match should be made
+      @ In, unpackXArray, bool, optional, True if the coordinates of the xarray variables must be exposed in the dict (e.g. if P(t) => {P:ndarray, t:ndarray})
+      @ Out, r, int, index where match was found OR size of data if not found
+      @ Out, rlz, dict, realization as {var:value} OR None if not found
+    """
+    assert(self._data is not None)
+    # TODO this could be slow, should do KD tree instead
+    mask = 1.0
+    for var,val in match.items():
+      # float instances are relative, others are absolute
+      if mathUtils.isAFloatOrInt(val):
+        # scale if we know how
+        try:
+          loc,scale = self._scaleFactors[var]
+        #except TypeError:
+        #  # self._scaleFactors is None, so set them
+        #  self._setScalingFactors(var)
+        except KeyError: # IndexError?
+        # variable doesn't have a scale factor (yet? Why not?)
+          loc = 0.0
+          scale = 1.0
+        scaleVal = (val-loc)/scale
+        # create mask of where the dataarray matches the desired value
+        mask *= abs((self._data[var]-loc)/scale - scaleVal) < tol
+      else:
+        mask *= self._data[var] == val
+    rlz = self._data.where(mask,drop=True)
+    try:
+      idx = rlz[self.sampleTag].item(0)
+    except IndexError:
+      return len(self),None
+    return idx,self._getRealizationFromDataByIndex(idx,unpackXArray)
+
+  def _getRequestedElements(self,options):
+    """
+      Obtains a list of the elements to be written, based on defaults and options[what]
+      @ In, options, dict, general list of options for writing output files
+      @ Out, keep, list(str), list of variables that will be written to file
+    """
+    if 'what' in options.keys():
+      elements = options['what'].split(',')
+      keep = []
+      for entry in elements:
+        small = entry.strip().lower()
+        if small == 'input':
+          keep += self._inputs
+          continue
+        elif small == 'output':
+          keep += self._outputs
+          continue
+        elif small == 'metadata':
+          keep += self._metavars
+          continue
+        else:
+          keep.append(entry.split('|')[-1].strip())
+    else:
+      # TODO need the sampleTag meta to load histories # BY DEFAULT only keep inputs, outputs; if specifically requested, keep metadata by selection
+      keep = self._inputs + self._outputs + self._metavars
+    return keep
+
+  def _getVariableIndex(self,var):
+    """
+      Obtains the index in the list of variables for the requested var.
+      @ In, var, st
```

Comment:
This is exactly the right place to put performance test results. Thank the stars that this code has been preserved in comment, since it doesn't exist anywhere else like a git history. /s

Reply:
Well, we all know how easy it is to run `git find --heuristic "that one place where I did the study and got some numbers and compared it with some other codes, somewhere near here or in this file I think" -f`